bears/general/FilenameBear.py

```diff
@@ -98,6 +98,11 @@ def run(self, filename, file,
                 'Filename does not use the suffix {!r}.'.format(
                     filename_suffix))
 
+        if len(new_name) > 4096-len(extension):
```

Comment:
Have you thought about making this a setting instead of hardcoding the value? Also what about the 260 char limit of windows? Shall the default limit be maybe system dependent?

Reply:
Additionally, you metioned about system dependent limit and I tried to check limits on Windows, Linux and OS X. I found that limit on Windows is really 260, limit on Linux is 4096 and also limit on OS X is different than previous metioned (probably 255). So do you think that better idea would be to check the current system or set default length as minimum of all systems?

Reply:
Probably, at least the default should may be variable. However for the sake of cross-compatibility this might be less good.

> OS X is different than previous metioned (probably 255)

Should be something Linux-like, as OSX is more or less BSD.

> I found that limit on Windows is really 260

And I believe it's possible to change the limit in Windows. 260 chars is for compatibility with older software (and sadly still default). Maybe we should detect the actual compatibility mode? Not sure, sounds like work 😅

Reply:
Should I consider that on Linux (and probably OSX) maximum length of filename is 255, however maximum path length is even 4096?

Reply:
Additionally, [here](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx#maxpath) is metioned that maximum path length is 260, however.. 
"The Windows API has many functions that also have Unicode versions to permit an extended-length path for a maximum total path length of 32,767 characters." 
Seems be that this is various in dependent of functions, however this can be hard to detection, so better choice should be take minimum, which is 260.

Reply:
> so better choice should be to take minimum, which is 260.

sounds good :)